msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/DiscussionForumTool.java

```diff
@@ -2224,6 +2236,220 @@ public String processActionToggleDisplayExtendedDescription()
+    String calendarBeginId;   //these values are used throughout the creation and maintenance of the Calendar sending.
+    String calendarEndId;
+    Boolean sendOpenCloseToCalendar;
+    Boolean availabilityRestrictedNow;
+    String title;
+    Date openDate;
+    Date closeDate;
+    String openingTitle;
+    String closingTitle;
+    Set membershipItems;
+    if (forumItem instanceof DiscussionTopic){  //this part of the processing determines if the MutableEntity is a Forum or a Topic and fills in all the variables by appropriately casting.
+      calendarBeginId = ((DiscussionTopic) forumItem).getCalendarBeginId();
+      calendarEndId = ((DiscussionTopic) forumItem).getCalendarEndId();
+      sendOpenCloseToCalendar = ((DiscussionTopic) forumItem).getSendOpenCloseToCalendar();
+      openDate = ((DiscussionTopic) forumItem).getOpenDate();
+      closeDate = ((DiscussionTopic) forumItem).getCloseDate();
+      membershipItems = ((DiscussionTopic) forumItem).getMembershipItemSet();
+      availabilityRestrictedNow = ((DiscussionTopic) forumItem).getAvailabilityRestricted();
+      openingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.topic.assembly",new Object[]{((DiscussionTopic) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.open")});
+      closingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.topic.assembly",new Object[]{((DiscussionTopic) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.close")});
+    } else if (forumItem instanceof DiscussionForum) {
+      calendarBeginId = ((DiscussionForum) forumItem).getCalendarBeginId();
+      calendarEndId = ((DiscussionForum) forumItem).getCalendarEndId();
+      sendOpenCloseToCalendar = ((DiscussionForum) forumItem).getSendOpenCloseToCalendar();
+      openDate = ((DiscussionForum) forumItem).getOpenDate();
+      closeDate = ((DiscussionForum) forumItem).getCloseDate();
+      membershipItems = ((DiscussionForum) forumItem).getMembershipItemSet();
+      availabilityRestrictedNow = ((DiscussionForum) forumItem).getAvailabilityRestricted();
+      openingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.forum.assembly",new Object[]{((DiscussionForum) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.open")});
+      closingTitle = getResourceBundleString("sendOpenCloseToCalendar.title.forum.assembly",new Object[]{((DiscussionForum) forumItem).getTitle(),getResourceBundleString("sendOpenCloseToCalendar.topic.close")});
+    } else {  //don't do anything with forumItem or the Calendar if it's not a Forum or a Forum Topic.
+      return;
+    }
+    Collection<Group> allowedGroups = getAllowedGroups(membershipItems);
+    if (allowedGroups.size() == 0 && !studentsAllowed(membershipItems)){
```

Comment:
Try and leave a space before your {. You do in some places, but don't in others. I like the space before the {, but I am a pedant.